src/server.h

```diff
@@ -3023,7 +3023,7 @@ void replicationStartPendingFork(void);
 void replicationHandlePrimaryDisconnection(void);
 void replicationCachePrimary(client *c);
 void resizeReplicationBacklog(void);
-void replicationSetPrimary(char *ip, int port);
+void replicationSetPrimary(char *ip, int port, int disconnect_blocked);
```

Comment:
IIUC, the only case we do not want to disco blocked clients is only in the case of  'server.failover_state = FAILOVER_IN_PROGRESS' right? can we then just check for this condition inside replicationSetPrimary instead of passing on an external condition?

Reply:
Yes, that should be possible, but I am reluctant to make `replicationSetPrimary` (which isn't specific to failover) behave differently based on some sub-state of the the failover state machine. I am not against it, but I think the intention gets clearer if the caller of `replicationSetPrimary`  states "I will take care of blocked clients myself".

Reply:
IMO passing this flag to the replicationSetPrimary and having it handle clients disconnects is already extending the function scope. I am fine with letting it handle the state logic and encapsulate the decision into a single place.

Reply:
I agree with @gmbnomis that we shouldn't introduce this dependency (`server.failover_state = FAILOVER_IN_PROGRESS`) to `replicationSetPrimary`. There's no inherent order between setting the failover state to in-progress and fixing the replication states. Introducing a failover state check adds an unrelated concern to the function, which in turn forces the caller to perform these operations in a specific sequence, an unnecessary constraint IMO and also not resilient to future changes.

Reply:
O.K. I agree this is also a good reasoning.